tests/components/mystrom/test_init.py

```diff
@@ -0,0 +1,171 @@
+    ):
+        config_entry = MockConfigEntry(
+            domain=DOMAIN,
+            entry_id=ENTRY_ID,
+            unique_id="uuid",
+            data={CONF_HOST: "1.1.1.1"},
+            title=DEVICE_NAME,
+        )
+        config_entry.add_to_hass(hass)
+
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert config_entry.state == ConfigEntryState.LOADED
+
+
+async def test_init_switch(hass: HomeAssistant) -> None:
+    """Test the initialization of a myStrom switch."""
+    await init_integration(hass, 101)
+    state = hass.states.get("switch." + DEVICE_NAME)
+    assert state is not None
+
+
+async def test_init_bulb(hass: HomeAssistant) -> None:
+    """Test the initialization of a myStrom bulb."""
+    await init_integration(hass, 102)
+    state = hass.states.get("light." + DEVICE_NAME)
+    assert state is not None
+
+
+async def test_init_of_unknown_bulb(hass: HomeAssistant) -> None:
+    """Test the initialization of a unknown myStrom bulb."""
+    with patch(
+        "pymystrom.get_device_info",
+        side_effect=AsyncMock(return_value={"type": 102, "mac": DEVICE_MAC}),
+    ), patch("pymystrom.bulb.MyStromBulb.get_state", return_value={}), patch(
+        "pymystrom.bulb.MyStromBulb.bulb_type", "new_type"
+    ), patch(
+        "pymystrom.bulb.MyStromBulb.mac",
+        new_callable=PropertyMock,
+        return_value=DEVICE_MAC,
+    ):
+        config_entry = MockConfigEntry(
+            domain=DOMAIN,
+            entry_id=ENTRY_ID,
+            unique_id="uuid",
+            data={CONF_HOST: "1.1.1.1"},
+            title="myStrom",
+        )
+        config_entry.add_to_hass(hass)
+
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+        assert config_entry.state == ConfigEntryState.SETUP_ERROR
+
+
+async def test_init_of_unknown_device(hass: HomeAssistant) -> None:
+    """Test the initialization of a unsupported myStrom device."""
+    with patch(
+        "pymystrom.get_device_info",
+        side_effect=AsyncMock(return_value={"type": 103, "mac": DEVICE_MAC}),
+    ):
+        config_entry = MockConfigEntry(
+            domain=DOMAIN,
+            entry_id=ENTRY_ID,
+            unique_id="uuid",
+            data={CONF_HOST: "1.1.1.1"},
+            title="myStrom",
+        )
+        config_entry.add_to_hass(hass)
+
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert config_entry.state == ConfigEntryState.SETUP_ERROR
+
+
+async def test_unload(hass: HomeAssistant) -> None:
+    """Test the unloading of a myStrom witch."""
+    await init_integration(hass, 101)
+    await hass.config_entries.async_unload(ENTRY_ID)
+    await hass.async_block_till_done()
+    state = hass.states.get("switch.test")
+    assert state is None
+
+
+async def test_init_cannot_connect(hass: HomeAssistant) -> None:
```

Comment:
I think this test should be split in two.
One for failed connect, and one for failed get_state.
```suggestion
async def test_init_cannot_connect(hass: HomeAssistant, config_entry: MockConfigEntry) -> None:
```